connectors/kafka/src/main/java/forklift/controller/KafkaController.java

```diff
@@ -212,39 +238,59 @@ private void controlLoop() {
         }
     }
 
-    private void commitPendingAndWaitForTopics() throws InterruptedException {
-        //check if the last remaining topic was removed
-        if (kafkaConsumer.assignment().size() > 0) {
-            Map<TopicPartition, OffsetAndMetadata>
-                            offsets =
-                            acknowledgmentHandler.removePartitions(kafkaConsumer.assignment());
+    private boolean processTopicChanges() throws InterruptedException {
+        boolean topicsChanged = processTopicAdd();
+        topicsChanged = processTopicRemoved() || topicsChanged;
```

Comment:
I see that you're trying to avoid short-circuit evaluation; I think this would look clearer as two variable declarations. E.g.

```java
boolean topicsAdded = processTopicAdd();
boolean topicsRemoved = processTopicRemoved();

if (topicsAdded || topicsRemoved) {
    kafkaConsumer.subscribe(topics, new RebalanceListener());
}
```

At the very least, I think a comment is warranted for clarity, since it isn't really intuitively obvious why the code is structured this way and, the `processTopicAdd()` and `processTopicRemoved()` methods do have symmetric roles.